histomicstk/features/ComputeHaralickFeatures.py

```diff
@@ -0,0 +1,210 @@
+        'Haralick.SumAverage.Range',
+        'Haralick.SumVariance.Range',
+        'Haralick.SumEntropy.Range',
+        'Haralick.Entropy.Range',
+        'Haralick.Variance.Range',
+        'Haralick.DifferenceEntropy.Range',
+        'Haralick.IMC1.Range',
+        'Haralick.IMC2.Range',
+    ]
+
+    # compute object properties if not provided
+    if rprops is None:
+        rprops = regionprops(im_label)
+
+    # create pandas data frame containing the features for each object
+    numFeatures = len(feature_list)
+    numLabels = len(rprops)
+    fdata = pd.DataFrame(np.zeros((numLabels, numFeatures)),
+                         columns=feature_list)
+
+    # check for consistent shapes between 'I' and 'Label'
+    if im_intensity.shape != im_label.shape:
+        raise ValueError("Inputs 'I' and 'Label' must have same shape")
+
+    # determine if image is grayscale or RGB
+    if len(im_intensity.shape) != 2:  # color image
+        raise ValueError("Inputs 'I' should be a grayscale image")
+
+    for i in range(numLabels):
+        # get bounds of an intensity image
+        minr, minc, maxr, maxc = rprops[i].bbox
+        # grab nucleus mask
+        subImage = im_intensity[minr:maxr+1, minc:maxc+1].astype(np.uint8)
+        # gets GLCM or gray-tone spatial dependence matrix
+        arrayGLCM = graycomatrixext(subImage, offsets=offsets,
+                                    num_levels=num_levels,
+                                    gray_limits=gray_limits)
+        num_dims = len(subImage.shape)
+        # offsets
+        if offsets is None:
+            # set default offset value
+            offsets = np.identity(num_dims)
+        else:
+            # check sanity
+            if offsets.shape[1] != num_dims:
+                raise ValueError(
+                    'Dimension mismatch between input image and offsets'
+                )
+        num_offsets = offsets.shape[0]
+        f = np.zeros((numFeatures/2, num_offsets))
+        for r in range(num_offsets):
+            # normalize GLCM
+            R = np.sum(arrayGLCM[:, :, r], dtype=np.float)
+            nGLCM = arrayGLCM[:, :, r]/R
+            # get marginal-probability matrix summing the rows
+            px = np.sum(nGLCM, axis=1)
+            py = np.sum(nGLCM, axis=0)
+            # initialize marginal-probability matrix sets
+            # summing normalizedGLCM such that i+j = k or i-j = k
+            pxPlusy = np.zeros(2*num_levels-1)
+            pxMinusy = np.zeros(num_levels)
+            # arbitarily small positive constant to avoid log 0
+            e = 0.00001
+            for n in range(0, num_levels):
+                for m in range(0, num_levels):
+                    # gets marginal-probability matrix
+                    pxPlusy[n+m] = pxPlusy[n+m] + nGLCM[n, m]
+                    pxMinusy[abs(n-m)] = pxMinusy[abs(n-m)] + nGLCM[n, m]
+            # f0: computes angular second moment
+            f[0, r] = np.sum(np.square(nGLCM))
```

Comment:
Use the form `fdata.at[i, Shape.Eccentricity]` (See [here](https://github.com/DigitalSlideArchive/HistomicsTK/blob/master/histomicstk/features/ComputeMorphometryFeatures.py#L117)) instead of `f[0, r], f[1, r], f[2, r] etc` to improve code readability. If i change the order of features in the feature list or add a new feature in the middle sometime later, then i will have to change a lot of indices which is prone to errors. Also it gets clumsy (i have to scroll up and see the feature list) to find which is the 0th, 1st, 2nd etc. feature. Always try to make the code easy to read for other developers, so that they can make modifications easily to it later when needed.